src/proxy/topic/ReliableTopicListenerRunner.ts

```diff
@@ -59,45 +59,55 @@ export class ReliableTopicListenerRunner<E> {
 
         this.ringbuffer.readMany(this.sequenceNumber, 1, this.batchSize)
             .then((result: ReadResultSet<ReliableTopicMessage>) => {
-                if (!this.cancelled) {
-                    for (let i = 0; i < result.size(); i++) {
-                        const msg = new Message<E>();
-                        const item = result.get(i);
-                        msg.messageObject = this.serializationService.toObject(item.payload);
-                        msg.publisher = item.publisherAddress;
-                        msg.publishingTime = item.publishTime;
-                        setImmediate(this.listener, msg);
-                        this.sequenceNumber++;
-                    }
-                    setImmediate(this.next.bind(this));
+                if (this.cancelled) {
+                    return;
                 }
-            })
-            .catch((e) => {
-                let message: string;
-                if (e instanceof StaleSequenceError) {
-                    this.ringbuffer.headSequence().then((seq: Long) => {
-                        const newSequence = seq.toNumber();
-
-                        message = 'Topic "' + this.proxy.getName() + '" ran into a stale sequence. ' +
-                            ' Jumping from old sequence ' + this.sequenceNumber + ' to new sequence ' + newSequence;
-                        this.logger.warn('ReliableTopicListenerRunner', message);
-
-                        this.sequenceNumber = newSequence;
-                        setImmediate(this.next.bind(this));
-                    });
 
-                    return;
+                const nextSeq = result.getNextSequenceToReadFrom().toNumber();
+                const lostCount = nextSeq - result.getReadCount() - this.sequenceNumber;
+                if (lostCount !== 0) {
+                    this.logger.trace('ReliableTopicListenerRunner', 'Listener of topic: ' + this.proxy.getName()
+                        + ' lost ' + lostCount + ' messages');
```

Comment:
Good catch. Changed the logic in 9d6b40d274bcbaf29f9d05a40009bdf2c807736b

Reply:
I'd like to reiterate on this one.

The previous implementation was handling `StaleSequenceError` as if it was based on a loss tolerant listener. I think we should do the same here. Otherwise we leave users sitting on a time bomb. If producers have a spike in message generation for a single time, the listener will be terminated and users will have no programmatic way to handle the termination and, say, register another listener. I have reverted the logic in cdcf57cfd71c9dc77ee9f9639be046f740d3b04e

WDYT?

Off-topic: I have also removed redundant `setImmediate` from `ReliableTopicListenerRunner`. See: 56a37ca07bf7a3f66474bd137c22b9239b63c565

Reply:
Hmm, okay, that makes sense. With a `warn` level log, I guess, it should be okay.